search_service/api/document.py

```diff
@@ -78,11 +79,15 @@ def put(self) -> Tuple[Any, int]:
         :param data: list of data objects to be indexed in Elasticsearch
         :return: name of index
         """
-        self.parser.add_argument('data', required=True)
+        self.parser.add_argument('data', required=True, action='append')
         args = self.parser.parse_args()
 
         try:
-            data = self.schema(many=True, strict=False).loads(args.get('data')).data
+            table_dict_list = []
+            for table_str in args.get('data'):
+                table_dict_list.append(literal_eval(table_str))
```

Comment:
is table_str actually a tablefields object? the naming is a bit confusing. also we typically don't put type behind the variable for naming.

Reply:
what are we using `literal_eval` here for?

Reply:
table_str is a string representation of dict, like '{"key1": val1, "key2": val2}'
`literal_eval` will concert the table_str into a dict first, or `json.dumps` will dump a list of `string` instead of a list of `dict`